disperser/dataapi/subgraph_client.go

```diff
@@ -106,6 +119,68 @@ func (sc *subgraphClient) QueryBatchNonSigningOperatorIdsInInterval(ctx context.
 	return batchNonSigningOperatorIds, nil
 }
 
+func (sc *subgraphClient) QueryNumBatchesByOperatorsInThePastBlockTimestamp(ctx context.Context, blockTimestamp uint64) (map[string]int, error) {
+	var (
+		registeredOperators   []*subgraph.Operator
+		deregisteredOperators []*subgraph.Operator
+		err                   error
+		pool                  = workerpool.New(maxWorkerPoolSize)
+	)
+
+	pool.Submit(func() {
+		operators, errQ := sc.api.QueryRegisteredOperatorsGreaterThanBlockTimestamp(ctx, blockTimestamp)
+		if errQ != nil {
+			err = errQ
+		}
+		registeredOperators = operators
+	})
+
+	pool.Submit(func() {
+		operators, errQ := sc.api.QueryDeregisteredOperatorsGreaterThanBlockTimestamp(ctx, blockTimestamp)
+		if errQ != nil {
+			err = errQ
+		}
+		deregisteredOperators = operators
+	})
+	pool.StopWait()
+
+	if err != nil {
+		return nil, err
+	}
+
+	intervalEvents, err := getOperatorsWithRegisteredDeregisteredIntervalEvents(ctx, registeredOperators, deregisteredOperators)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		mu                   sync.Mutex
+		numBatchesByOperator = make(map[string]int, 0)
+		intervalEventsPool   = workerpool.New(maxWorkerPoolSize)
+	)
+	for ie := range intervalEvents {
+		interval := ie
+		intervalEventsPool.Submit(func() {
+			end := interval.Events[1]
+			if end == 0 {
+				end = uint64(time.Now().Unix())
+			}
+			batches, err := sc.api.QueryBatchesByBlockTimestampRange(ctx, interval.Events[0], end)
```

Comment:
you need to filter by reference block number, not timestamp of the batch confirmation

Reply:
If the block time is monotonically increasing w.r.t. the block number, it should produce equivalent results?
i.e. for block A and B, blockTime(A) < blockTime(B) iff blockNumber(A) < blockNumber(B); if this is true, querying the interval by block number will be equivalent to querying the interval by block time.